File: temas/herencia/ejemplo.java
public class ejemplo {
    public static void main(String[] args) {
        Animal miAnimal = new Animal("Mimi", 2);
        Perro miPerro = new Perro("Fido", 3, "Labrador");
        
        miAnimal.hacerSonido();
        miPerro.hacerSonido();
        miPerro.ladrar();
    }
}

File: temas/herencia/ejemplo.scala
object ejemplo {
    def main(args: Array[String]): Unit = {
    val miAnimal = new Animal("Mimi", 2)
    val miPerro = new Perro("Fido", 3, "Labrador")
    miAnimal.hacerSonido()
    miPerro.hacerSonido()
    miPerro.ladrar()
    }
}